src/main/scala/za/co/absa/abris/avro/parsing/utils/AvroSchemaUtils.scala

```diff
@@ -82,9 +82,9 @@ object AvroSchemaUtils {
     configureSchemaManager(schemaRegistryConf)
 
     val subject = SchemaManager.getSubjectName(topic, isKey, schema, schemaRegistryConf)
-    if (!SchemaManager.exists(subject) || SchemaManager.isCompatible(schema, subject)) {
+    if (subject.isDefined && (!SchemaManager.exists(subject.get) || SchemaManager.isCompatible(schema, subject.get))) {
```

Comment:
I would maybe rename `val subject` to `val subjectOpt` and swap out this if else with pattern matching, something like:
```
subjectOpt match {
  case Some(subject) =>
    if-else as before
  case None                 =>
    logger.error("Subject undefined")
```

Reply:
Not sure how this would play with the error log from `SchemaManager.getSubjectName`, just a thought.